File: elixir/cancel_domain_transfer.exs
# Usage: mix run cancel_domain_transfer.exs example.com 42
# where 42 is the domain transfer id

client = %Dnsimple.Client{
  access_token: Application.get_env(:dnsimple, :access_token),
  base_url: "https://api.sandbox.dnsimple.com/"
}

{:ok, response} = Dnsimple.Identity.whoami(client)
account_id = response.data.account.id

[name, transfer_id] = System.argv

IO.puts "Cancelling Transfer #{transfer_id} for domain #{name}"

IO.inspect Dnsimple.Registrar.cancel_domain_transfer(client, account_id, name, transfer_id)

File: node/cancel_domain_transfer.js
const { DNSimple, AuthenticationError } = require('dnsimple');

(async () => {
  const dnsimple = new DNSimple({
    baseUrl: 'https://api.sandbox.dnsimple.com',
    accessToken: process.env.TOKEN,
    userAgent: 'dnsimple-examples',
  });

  try {
    const identity = await dnsimple.identity.whoami();
    const accountId = identity.data.account.id;
    const domainName = process.env.DOMAIN;
    const transferId = process.env.TRANSFER_ID;

    if (!domainName) {
      console.error('Please specify a domain name to register');
      return;
    }

    if (!transferId) {
      console.error('Please specify a transfer ID to cancel the transfer with');
      return;
    }

    const transfer = await dnsimple.registrar.cancelDomainTransfer(accountId, domainName, transferId);
    console.log(transfer);
  } catch (err) {
    if (err instanceof AuthenticationError) {
      console.error('Authentication error. Check your token is correct for the sandbox environment.');
    } else {
      console.error(err);
    }
  }
})();
